src/realm/util/logger.hpp

```diff
@@ -77,32 +77,51 @@ class Logger {
     // equivalent to realm_log_level_e in realm.h and must be kept in sync
     enum class Level { all = 0, trace = 1, debug = 2, detail = 3, info = 4, warn = 5, error = 6, fatal = 7, off = 8 };
 
+    static const Level default_log_level;
+
     template <class... Params>
     void log(Level, const char* message, Params&&...);
 
-    virtual Level get_level_threshold() noexcept
+    virtual Level get_level_threshold() const noexcept
     {
-        return m_level_threshold;
+        // Don't need strict ordering, mainly that the gets/sets are atomic
+        return m_level_threshold.load(std::memory_order_relaxed);
     }
 
     virtual void set_level_threshold(Level level) noexcept
     {
-        m_level_threshold = level;
+        // Don't need strict ordering, mainly that the gets/sets are atomic
+        m_level_threshold.store(level, std::memory_order_relaxed);
     }
 
     /// Shorthand for `int(level) >= int(m_level_threshold)`.
     inline bool would_log(Level level) const noexcept
     {
-        return static_cast<int>(level) >= static_cast<int>(m_level_threshold);
+        return static_cast<int>(level) >= static_cast<int>(get_level_threshold());
     }
 
     virtual inline ~Logger() noexcept = default;
 
 protected:
-    Logger() noexcept = default;
+    Logger() noexcept
+        : m_threshold_ptr{std::make_unique<std::atomic<Level>>(default_log_level)}
+        , m_level_threshold{*m_threshold_ptr}
+    {
+    }
+
+    explicit Logger(Level level) noexcept
+        : m_threshold_ptr{std::make_unique<std::atomic<Level>>(level)}
```

Comment:
Why do we need to store this in a unique_ptr like this? Is it because otherwise Logger becomes unmovable?

Reply:
Moved to a regular object.